src/herder/HerderSCPDriver.cpp

```diff
@@ -933,6 +945,48 @@ HerderSCPDriver::recordSCPEvent(uint64_t slotIndex, bool isNomination)
     }
 }
 
+void
+HerderSCPDriver::recordSCPExternalizeEvent(uint64_t slotIndex, NodeID const& id,
+                                           bool forceupdateself)
+{
+    auto& timing = mSCPExecutionTimes[slotIndex];
+    auto now = mApp.getClock().now();
+
+    bool first = !timing.mFirstExternalize;
+
+    if (first)
+    {
+        timing.mFirstExternalize = make_optional<VirtualClock::time_point>(now);
+    }
+
+    bool self = (id == mSCP.getLocalNodeID());
+    if (self)
+    {
+        if (!timing.mSelfExternalize)
+        {
+            if (timing.mFirstExternalize)
```

Comment:
`timing.mFirstExternalize` is either present already or set on line 959, so I don't think you need this check.

Reply:
yeah I wanted to use `first` here (so bug), but I think it's better to always capture. so I removed this condition